generator/src/main/java/com/graphicsfuzz/generator/mutateapi/ExpressionGenerator.java

```diff
@@ -0,0 +1,629 @@
+   * The binary expression adding the remainder to the return value will also be
+   * injected if there is a left over from a division of original value by the divisor.
+   *
+   * <p>For example, given x value we can derive a for loop statement and a remainder from the
+   * equation x = (divisor * iterations) + remainder.
+   *
+   * <p>The code fragment below is an example result obtained by this method which returns an int 7.
+   * int _GLF_PRIMITIVE_int_7 = 0;
+   * // 3 is the number of iterations obtained by dividing the given value 7 by the random
+   * divisor 2.
+   * for(int i = 0; i < 3; i ++) {
+   * _GLF_PRIMITIVE_int_7 += 2;   // 2 is a divisor which is randomly generated.
+   * }
+   * _GLF_PRIMITIVE_int_7 += 1;     // 1 is a remainder after dividing 7 by 2.
+   *
+   * @param factManager        manager class holding the value and its associated expression that
+   *                           guarantees to compute the given value.
+   * @param functionDefinition a function into which the new expression will be injected.
+   * @param stmt               statement in the body of the given function where the generated
+   *                           expression will be inserted before.
+   * @param value              the value that will be computed by the expression generated by
+   *                           this method.
+   * @return a variable identifier to the new variable generated by this method.
+   */
+
+  private Expr generateForLoopValue(FactManager factManager,
+                                    FunctionDefinition functionDefinition,
+                                    Stmt stmt, Value value) {
+    if (value.getType() != BasicType.INT) {
+      return null;
+    }
+    // If the global fact manager is given, the caller expect the generated expression to be
+    // available at the global scope. However the new for loop statement generated by this
+    // method will be injected into the body of a given function making it available only in
+    // the local scope. Hence, if the global scope expression is requested, null is returned.
+    if (factManager.globalScope()) {
+      return null;
+    }
+
+    // Declare and initialize a new zero-value variable.
+    final String varName = genVarName(value, false);
+    final VariableDeclInfo variableDeclInfo = new VariableDeclInfo(varName, null,
+        new Initializer(
+            generateExpr(factManager,
+                functionDefinition,
+                stmt,
+                new NumericValue(BasicType.INT, Optional.of(0)))
+        ));
+    final VariablesDeclaration variablesDecl = new VariablesDeclaration(value.getType(),
+        variableDeclInfo);
+    functionDefinition.getBody().insertBefore(stmt, new DeclarationStmt(variablesDecl));
+
+    // Decide whether the loop should be incremented or decremented for each iteration.
+    final boolean isIncrement = generator.nextBoolean();
+    // If value is unknown, we could generate and return any number.
```

Comment:
A concern here: if we choose a very large integer we may end up with a loop with masses of iterations.

Can you open an issue for that with a TODO that links to the issue?

Reply:
Thanks for opening the issue; please add:

// TODO(github issue URL): (very brief summary of issue)